routes/auth.js

```diff
@@ -0,0 +1,64 @@
+const express = require('express')
+const bcrypt = require('bcryptjs')
+const jwt = require('jsonwebtoken')
+const User = require('../models/User')
+const auth = require('../middleware/auth')
+
+const router = express.Router()
+
+const { JWT_SECRET } = process.env
+
+router.post('/', async (request, response, next) => {
+  const { username, password } = request.body
+
+  try {
+    let user = await User.findOne({ username })
+
+    if (!user) {
+      const error = { message: 'Username does not exist!' }
+      return next({ status: 400, message: error.message })
+    }
+
+    bcrypt.compare(password, user.password, (errCompare, resCompare) => {
+      if (errCompare) {
+        console.error(errCompare)
+        return next({ status: 400, message: errCompare.message })
+      }
+      if (resCompare) {
+        const payload = {
+          user: {
+            id: user._id,
+          },
+        }
+
+        // TODO: expiration to 3600
+        jwt.sign(
+          payload,
+          JWT_SECRET,
+          { expiresIn: 360000 },
+          (errJWT, token) => {
+            if (errJWT) {
+              console.error(errJWT)
+              return next({ status: 400, message: errJWT.message })
+            }
+            response.status(200).json(token)
+          }
+        )
+      } else {
+        const error = { message: 'Passwords do not match!' }
+        console.error(error.message)
+        return next({ status: 400, message: error.message })
+      }
+    })
+  } catch (err) {
+    console.error(err)
+    const error = { message: 'Unknown error!' }
+    return next({ status: 400, message: error.message })
+  }
```

Comment:
If you don't know what went wrong don't respond with 400 (4xx tells the client she did something wrong)